src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Application/Commands/UpdateApprenticeshipStopDateCommandHandlerTests.cs

```diff
@@ -0,0 +1,387 @@
+            // Arrange
+            var apprenticeship = await SetupApprenticeship(paymentStatus: PaymentStatus.Withdrawn, startDate: DateTime.UtcNow.AddMonths(2));            
+            var command = new UpdateApprenticeshipStopDateCommand(apprenticeship.Cohort.EmployerAccountId, apprenticeship.Id, DateTime.UtcNow, new UserInfo());
+
+            // Act
+            var exception = Assert.ThrowsAsync<DomainException>(async () => await _handler.Handle(command, new CancellationToken()));
+
+            // Assert
+            exception.DomainErrors.Should().BeEquivalentTo(new { PropertyName = "newStopDate", ErrorMessage = $"Invalid Date of Change. Date cannot be before the training start date." });
+        }    
+
+        [Test]
+        public async Task Handle_WhenHandlingCommand_WithValidateEndDateOverlap_ThenShouldThrowDomainException()
+        {
+            // Arrange
+            var apprenticeship = await SetupApprenticeship(paymentStatus: PaymentStatus.Withdrawn);
+            apprenticeship.Uln = "X";
+            var command = new UpdateApprenticeshipStopDateCommand(apprenticeship.Cohort.EmployerAccountId, apprenticeship.Id, DateTime.UtcNow, new UserInfo());
+            _overlapCheckService.Setup(x => x.CheckForOverlaps(It.Is<string>(uln => uln == "X"), It.IsAny<CommitmentsV2.Domain.Entities.DateRange>(), It.IsAny<long?>(), It.IsAny<CancellationToken>()))
+             .ReturnsAsync(() => new OverlapCheckResult(false, true));
+
+            // Act
+            var exception = Assert.ThrowsAsync<DomainException>(async () => await _handler.Handle(command, new CancellationToken()));
+
+            // Assert
+            exception.DomainErrors.Should().BeEquivalentTo(new { PropertyName = "StopDate", ErrorMessage = $"The date overlaps with existing dates for the same apprentice." + Environment.NewLine + "Please check the date - contact the provider for help" });
+        }
+
+        [Test]
+        public async Task Handle_WhenHandlingCommand_UpdateApprenticeshipStopDate_ThenShouldUpdateDatabaseRecord()
+        {
+            // Arrange
+            var apprenticeship = await SetupApprenticeship(paymentStatus: PaymentStatus.Withdrawn);
+            var newStopDate = new DateTime(DateTime.UtcNow.Year, DateTime.UtcNow.Month, 1);
+
+            var command = new UpdateApprenticeshipStopDateCommand(apprenticeship.Cohort.EmployerAccountId, apprenticeship.Id, newStopDate,  new UserInfo());
+
+            // Act
+            await _handler.Handle(command, new CancellationToken());
+            // Simulate Unit of Work contex transaction ending in http request.
+            await _dbContext.SaveChangesAsync();
+
+            // Assert
+            var apprenticeshipAssertion = await _confirmationDbContext.Apprenticeships.FirstAsync(a => a.Id == apprenticeship.Id);
+            apprenticeshipAssertion.StopDate.Should().Be(newStopDate);            
+            apprenticeshipAssertion.PaymentStatus.Should().NotBe(PaymentStatus.Completed);
```

Comment:
should this assert check that `PaymentStatus.Equal(PaymentStatus.Withdrawn)`

Reply:
done